Challenge list and leaderboard and minor changes on individual leaderboard

sunil07t opened the pull request:
- Challenge list divided into joined and more challenges sections

- Minor change on individual leaderboard to not display mount for user not in top 3. The card is to small to include Mount.

- CSS change to display gray color popup title

- The individual leaderboard is only loaded when the user goes into the leaderboard screen, removed individual leaderboard from refresh and getUserInfo.

AmplabJenkins commented:
Can one of the admins verify this patch?

shankari commented on bin/sign_and_align_keys.sh:
```diff
@@ -23,4 +23,4 @@ jarsigner -verbose -sigalg SHA1withRSA -digestalg SHA1 -keystore ~/Safety_Infras
 
 # Re-add the plugin
-ionic plugin add cordova-plugin-crosswalk-webview
+ionic plugin add cordova-plugin-crosswalk-webview
```
Can you revert the files that have no significant changes?
Like this one...

shankari commented on bower.json:
```diff
@@ -27,4 +27,4 @@
     "jquery": "^3.1.0"
   }
-}
+}
```
and this

shankari commented on config.xml:
```diff
@@ -84,4 +84,4 @@
     <variable name="SENDER_ID" value="97387382925" />
   </plugin>
-</widget>
+</widget>
```
and this...

shankari commented on www/css/style.css:
```diff
@@ -1717,9 +1724,9 @@ html, body, #chart2, svg {
-    overflow: scroll !important;
-}
+#iframe {   
+  /*width: 375px !important;*/    
```
Is this really a change?

shankari commented on www/templates/diary/list.html:
```diff
@@ -18,12 +18,12 @@
         on-refresh="refresh()">
         </ion-refresher>
-		<ion-list>
+        <ion-list>
```
no changes in this, I think

shankari commented on www/templates/goals/challenge-detail.html:
```diff
@@ -0,0 +1,35 @@
+<ion-modal-view style="background-color: #ededed;">
```
like the separate files for screens. Would be good to have matching separate js as well instead of dumping everything into `goals.js`

shankari commented on www/templates/goals/leaderboard-modal.html:
```diff
@@ -109,10 +109,10 @@ <h1 class = "title">Leaderboard</h1>
         <div ng-repeat="user in usersUnderTopThree" 
            class="card item-thumbnail-left-avatar leaderboard-card" id="avatar" style="padding-top: 10px !important;height: 117px !important;">
-          <figure class="herobox background_{{user.preferences.background}} true ? &quot;isUser&quot; : &quot;&quot;} undefined ? &quot;minimal&quot; : &quot;&quot;}" ng-class="{hasPet: (true &amp;&amp; user.items.currentPet), hasMount: (true &amp;&amp; user.items.currentMount), noBackgroundImage: !user.preferences.background, &quot;cast-target&quot;: applyingAction, isLeader: party.leader==user. ...
+          <figure class="herobox background_{{user.preferences.background}} true ? &quot;isUser&quot; : &quot;&quot;} undefined ? &quot;minimal&quot; : &quot;&quot;}" ng-class="{hasPet: (true &amp;&amp; user.items.currentPet), noBackgroundImage: !user.preferences.background, &quot;cast-target&quot;: applyingAction, isLeader: party.leader==user._id}" style="top: -10px !important; left: 0px !import ...
```
was this intentional? you don't want the `hasMount` check any more?

sunil07t replied:
Yes, this is for the all the users not in top 3 position. The card is to small to include the avatar, the pet and the mount so this version only displays the avatar and the pet.

shankari commented on www/templates/goals/leaderboard-modal.html:
```diff
@@ -109,10 +109,10 @@ <h1 class = "title">Leaderboard</h1>
             <div class="character-sprites">
               <!-- Mount Body-->
-              <span ng-if="user.items.currentMount" class="Mount_Body_{{user.items.currentMount}}"></span>
+              <!--span ng-if="user.items.currentMount" class="Mount_Body_{{user.items.currentMount}}"></span-->
```
removing more mount-related information?

shankari commented on www/templates/goals/leaderboard-modal.html:
```diff
@@ -142,7 +142,7 @@ <h1 class = "title">Leaderboard</h1>
               </span>
               <!-- Mount Head-->
-              <span ng-if="user.items.currentMount" class="current-mount Mount_Head_{{user.items.currentMount}}"></span>
+              <!--span ng-if="user.items.currentMount" class="current-mount Mount_Head_{{user.items.currentMount}}"></span-->
```
like this?

shankari commented on www/templates/main-goals.html:
```diff
@@ -212,18 +212,24 @@ <h5 style="font-weight:700;color:#6e6e6e;">{{bossName}}</h5>
           </span>
         </a>
+        <a id="challenge" class="col col-25 " style="text-align: center;color:#6e6e6e;margin-top: 5px;margin-bottom: 5px;" ng-click="openChallenge()">
+          <i class="icon ion-person-stalker" style="display: block;font-size: 20px;"></i>
```
interesting to see that the icon is named `ion-person-stalker`!

shankari commented on www/js/goals.js:
```diff
@@ -869,9 +857,9 @@ angular.module('emission.main.goals',['emission.services', 'emission.plugin.logg
-        getChallenges();
-    }
+    //if (storage.get(HABITICA_REGISTERED_KEY) == true) {
+       // getChallenges();
```
so now we get challenges always, not just when the user is registered into habitica?

sunil07t replied:
Yes, I made this change because now the getChallenges is called in the getUserInfo to get challenges every time when the page is refreshed. getUserInfo only works if the user is registered into habitica.

shankari commented on www/templates/main-goals.html:
```diff
@@ -212,18 +212,24 @@ <h5 style="font-weight:700;color:#6e6e6e;">{{bossName}}</h5>
       <div>
       <div class="row" style="border-top-style: solid;border-top-color: #ddd;border-top-width: 0.5px;border-bottom-style: solid;border-bottom-color: #ddd;border-bottom-width: 0.5px;">
-        <a id="leaderboard" class="col col-33" style="text-align: center;color:#6e6e6e;margin-top: 5px;margin-bottom: 5px;" ng-click="openLeaderboard()">
+       <a id="leaderboard" class="col col-33" style="text-align: center;color:#6e6e6e;margin-top: 5px;margin-bottom: 5px;" ng-click="openLeaderboard()">
```
why is this still `col-33`? Weren't you going to make them all be `col-25`?

shankari commented on www/js/goals.js:
```diff
@@ -689,117 +745,6 @@ angular.module('emission.main.goals',['emission.services', 'emission.plugin.logg
     };
 
-    /*var getUserChallenges = function(){
-        for(var challenge in $scope.joinedChallenges){
```
thanks for removing commented out code...

shankari commented on www/js/goals.js:
```diff
@@ -135,43 +151,55 @@ angular.module('emission.main.goals',['emission.services', 'emission.plugin.logg
     };
-
+    $scope.openChallenge = function() {
+        $scope.challengeModal.show();
```
we should be able to restructure this to a separate service later.
check out https://github.com/e-mission/e-mission-data-collection/tree/master/www/ui/ionic
for an example

sunil07t commented:
Sorry, I forgot to push the changes. New changes:

- Fixed buttons col space

- Load individual leaderboard is only when the user goes into the leaderboard screen, removed individual leaderboard from refresh and getUserInfo. This was slowing down other functions those use refresh like score up, join challenge.

sunil07t commented on www/js/goals.js:
```diff
@@ -196,13 +224,21 @@ angular.module('emission.main.goals',['emission.services', 'emission.plugin.logg
         var regConfig = {'username': $scope.theUser.username};
         console.log(regConfig);
         $ionicLoading.show({
-            template: '<ion-spinner icon="bubbles" class="costume"></ion-spinner>'
```
Yes, the costume class is not in the style, the loading screen is the same without this code.
